flake.nix

```diff
@@ -252,12 +300,48 @@
           git
           nil
           nixpkgs-fmt
+          toml2json
           llvmPackages.lldb # This ensures the right lldb is in the environment for running rust-lldb
+          wasm-bindgen-cli
         ];
 
         shellHook = ''
+          echo which wasm-bindgen $(which wasm-bindgen)
           eval "$(starship init bash)"
         '';
       });
+
+      packages.wasm = craneLib.mkCargoDerivation (noirWasmArgs // {
+
+        inherit GIT_COMMIT;
+        inherit GIT_DIRTY;
+        doCheck = false;
+
+        cargoArtifacts = noir-wasm-cargo-artifacts;
+        
+        COMMIT_SHORT = builtins.substring 0 7 GIT_COMMIT;
+        VERSION_APPENDIX = if GIT_DIRTY == "true" then "-dirty" else "";
+        PKG_PATH = "./pkg";
+
+        nativeBuildInputs = with pkgs; [
+          which
+          git
+          jq
+          rustToolchain
+          wasm-bindgen-cli
+          binaryen
+          toml2json
+        ];
+
+        buildPhaseCargoCommand = ''
```

Comment:
should be separated into `cargoExtraArgs` and a `postBuild`

Reply:
Separated into cargoExtraArgs and postBuild. Also separated the buildPhaseCargoCommand and added a preBuild for the step of deleting ./pkg/

Reply:
Do we need to delete the `./pkg/` directory since nix builds in fresh sandboxes? I think that might just be a holdover from before nix

Reply:
we need to delete it if we're not in nix context and build process is not managed by nix? Remeber our contract was to use nix but also allow to escape it whenever possible... so we should be able to build w/o nix just the same...